southbound/impl/src/main/java/org/opendaylight/ansible/southbound/AnsibleCommandServiceImpl.java

```diff
@@ -69,14 +88,30 @@ public AnsibleCommandServiceImpl(@OsgiService final DataBroker dataBroker) {
                 .setFailedEvent(failedEvent).build()).buildFuture();
     }
 
-    private Uuid runAnsible(String host, String dir, String file) throws ManagedProcessException {
-        Uuid uuid = new Uuid(UUID.randomUUID().toString());
-        LOG.info("Executing Ansible, new uuid for command is: " + uuid);
+    private Uuid runAnsibleRole(String host, String dir, String role, List<String> roleVars, List<String> ansibleVars)
+            throws ManagedProcessException {
+        ManagedProcessBuilder ar = new ManagedProcessBuilder("ansible-runner").addArgument("-j")
+                .addArgument("--hosts").addArgument(host).addArgument("-r").addArgument(role);
+        ar = injectVars(ar, roleVar, roleVars);
+        ar = injectVars(ar, ansibleVar, ansibleVars);
+        ar.addArgument("run").addArgument(dir);
+        return runAnsible(ar);
+    }
+
+    private Uuid runAnsiblePlaybook(String host, String dir, String file, List<String> ansibleVars)
+            throws ManagedProcessException {
         ManagedProcessBuilder ar = new ManagedProcessBuilder("ansible-runner").addArgument("-j")
                 .addArgument("--hosts").addArgument(host).addArgument("-p").addArgument(file);
+        ar = injectVars(ar, ansibleVar, ansibleVars);
         ar.addArgument("run").addArgument(dir);
-        ar.setProcessListener(new AnsibleProcessListener(this, uuid));
-        ManagedProcess mp = ar.build();
+        return runAnsible(ar);
+    }
+
+    private Uuid runAnsible(ManagedProcessBuilder builder) {
```

Comment:
in the old code it could still bubble up because the ManagedProcessException could be thrown by the builder. Since the builder is moved out of this method it doesnt throw it anymore.

Reply:
there is a try/catch in this method that catches ManagedPRocessException still right? If that catches, the method still returns a uuid and then success. So it doesn't bubble up.

Reply:
right, im just saying this method used to throw the exception and now it doesnt, because in the old code it could throw the exception outside of the try clause. We catch the one in the try clause on purpose so all good.

Reply:
the problem is if you catch the exception, then return the uuid, then you always return success and InProgress to the rpc.

Reply:
yes it always returns in progress at first, but it will still fail in the callback. We can talk about changing that behavior (probably best not in this PR, since this patch doesnt change that code). There was a reason I think I did it that way, but we can talk about it later.